beesdoo_base/models/partner.py

```diff
@@ -49,19 +51,15 @@ def write(self, values):
                 if command[0] == 2:
                     command[0] = 3
         return super(Partner, self).write(values)
+    
+    @api.multi
+    def _deactivate_active_cards(self):
+        if len(self.member_card_ids) > 0:
+            for c in self.member_card_ids:
+                if c.valid:
+                    c.valid = False
+                    c.end_date = fields.Date.today()      
+    @api.multi            
+    def _new_card(self, txt):
+        self.env['member.card'].create({'partner_id' : self.env.context['active_id'],'comment' : txt})
```

Comment:
can use self.env['member.card'].create({'partner_id' : self.env.context['active_id'],'comment' : txt})
self.id is already the record of the partner don't need to use self.env.context['active_id']

if it's @api.multi you need to iterate on self can use @api.one instead